jwst/outlier_detection/outlier_detection.py

```diff
@@ -119,10 +119,10 @@ def do_detection(self):
             sdriz.do_drizzle(**pars)
             drizzled_models = sdriz.output_models
             for model in drizzled_models:
-                model.meta.filename += ".fits"
-            if save_intermediate_results:
-                log.info("Writing out resampled exposures...")
-                drizzled_models.save()
+                model.meta.filename += "_i2d.fits"
```

Comment:
Is it safe to just append the new suffix, rather than do a replacement of the old suffix? Or do you know that meta.filename will only ever be a "bare" base name?

Reply:
The file naming in outlier detection needs to be refactored.  Right now, a bunch of naming decisions (and appropriate suffixes) get done in subroutines or subroutines of subroutines, and really, they should all be happening in `OutlierDetectionStep.process` using the standard `Step` methods.

It's a bit messy, and the refactor will take some effort, so I would suggest we do this when we have some spare time in the near future.

Reply:
Agreed. If it ain't broke (right now), don't fix it (in B7.1). But be sure to keep an issue open so that we don't forget about it.

Reply:
I will open an issue then.

I've already refactored outlier_detection once.  Always fun to do it again.  Maybe the 2nd time we can do it with propagated error arrays.

Reply:
👍 for propagated error arrays